src/Integration.Vsix/SonarLintTagger/TextBufferIssueTracker.cs

```diff
@@ -274,11 +275,21 @@ void IIssueConsumer.Accept(string path, IEnumerable<Issue> issues)
                 return;
             }
 
-            // TODO: collect the additional information required to compare be able to filter
-            // the issues (i.e. the text of the line and the line hash)
-            // issues = issuesFilter.Filter(path, issues);
+            // Collect the additional information required to compare be able to filter
+            // the issues, then apply the filter
+            var unfilteredIssues = issues.Select(x => CreateFilterableIssue(x, currentSnapshot))
+                .Where(i => i != null)
+                .ToArray();
+
+            var filteredIssues = issuesFilter.Filter(path, unfilteredIssues);
+            Debug.Assert(filteredIssues.All(x => x is DaemonIssueAdapter), "Not expecting the issue filter to change the list item type");
 
-            var newMarkers = issues.Where(IsValidIssueTextRange).Select(CreateIssueMarker);
+            var suppressedCount = unfilteredIssues.Length - filteredIssues.Count();
+            logger.WriteLine(Strings.Daemon_SuppressedIssuesInfo, suppressedCount);
+
+            var newMarkers = filteredIssues.OfType<DaemonIssueAdapter>()
```

Comment:
I think this method is getting too big - I suggest extracting this to a method, something like:
IEnumerable<Issue> validIssues = GetNonSuppressedIssues(..)
var newMarkers = validIssues.Where(IsValidIssueTextRange).Select(CreateIssueMarker);